lib/ansible/modules/clustering/k8s/k8s_service.py

```diff
@@ -0,0 +1,618 @@
+          returned: success
+          type: string
+        load_balancer_source_ranges:
+          description:
+            - When the service is a load balancer, inbound traffic flow will be restricted to this set of IP
+            - ranges.
+          returned: success
+          type: string
+        ports:
+          description: The list of ports exposed by this service.
+          returned: success
+          type: complex
+        selector:
+          description:
+            - Key, value pairs used to determine which pods to route traffic to.
+          returned: success
+          type: complex
+        session_affinity:
+          description:
+            - Contains either I(ClientIp) or I(None), indicating which form of session affinity is
+            - active, if any.
+          returned: success
+          type: string
+        type:
+          description:
+            - Determines how the Service is exposed. Valid options are I(ExternalName), I(ClusterIP), I(NodePort),
+            - and I(LoadBalancer).
+          returned: success
+          type: string
+'''
+
+
+def create_service(module, kube):
+    '''
+    Create a new service.
+    Assembles a k8s template from module params, and posts it to the API.
+    :param module: Ansible module object
+    :param kube: kubernetes client
+    :param check_mode: bool
+    :return: API response JSON
+    '''
+
+    template = {
+        u'apiVersion': u'v1',
+        u'kind': u'Service',
+        u'metadata': {},
+        u'spec': {}
+    }
+
+    template['metadata']['name'] = module.params['name']
+    if module.params.get('labels'):
+        template['metadata']['labels'] = module.params.get('labels')
+    if module.params.get('annotations'):
+        template['metadata']['annotations'] = module.params.get('annotations')
+    if module.params.get('cluster_ip'):
+        template['spec']['clusterIP'] = module.params.get('cluster_ip')
+    if module.params.get('external_i_ps'):
+        template['spec']['externalIPs'] = module.params.get('external_i_ps')
+    if module.params.get('type'):
+        template['spec']['type'] = module.params.get('type')
+    if module.params.get('external_name'):
+        template['spec']['externalName'] = module.params.get('external_name')
+    if module.params.get('load_balancer_source_ranges'):
+        template['spec']['loadBalancerSourceRanges'] = module.params.get('load_balancer_source_ranges')
+    if module.params.get('ports'):
+        template['spec']['ports'] = convert_port_names(module.params['ports'])
+    if module.params.get('selector'):
+        template['spec']['selector'] = module.params.get('selector')
+    if module.params.get('session_affinity'):
+        template['spec']['sessionAffinity'] = module.params.get('session_affinity')
+
+    api_response = template
+    if not module.check_mode:
+        try:
+            api_response = kube.create_namespaced_service(module.params['namespace'], template)
```

Comment:
I think having a parent class in module_utils for each of namespaced and non-namespaced services would make sense, which namespace handling happening at that level.